### PR TITLE
Fix various

### PR DESCRIPTION
**Pull request by ZakKemble**
## Description

Some fixes for invalid HTML (using https://validator.w3.org/ to check), pagination fix when using a different path config and fix hidden posts still showing in archives.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How has this been tested

- [ ] I have run `bash ./tools/deploy.sh --dry-run` (at the root of the project) locally and passed
- [x] I have tested this feature in the browser

### Test Configuration

- Browser type & version: Firefox 98.0.2
- Operating system: Windows 10
- Ruby version: 2.7.4p191
- Bundler version: 2.3.8
- Jekyll version:  4.2.1

### Checklist

- [x] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: _includes/post-nav.html [hunk 1]
```diff
@@ -21,10 +21,10 @@
     <p>{{ page.next.title }}</p>
   </a>
   {% else %}
-  <span class="btn btn-outline-primary disabled"
+  <a class="btn btn-outline-primary disabled"
     prompt="{{ site.data.locales[lang].post.button.next }}">
     <p>-</p>
-  </span>
+  </a>
```
Comment (cotes2020) [inline _includes/post-nav.html:27]: No need for `<a>` here, please use `<div>` instead

File: _includes/post-paginator.html [hunk 2]
```diff
@@ -47,7 +47,7 @@
     {% if show %}
       <!-- show number -->
       <li class="page-item {% if i == paginator.page %} active{% endif %}">
-        <a class="page-link btn-box-shadow" href="{{ site.baseurl }}/{% if i > 1%}page{{ i }}/{% endif %}">{{ i }}</a>
+        <a class="page-link btn-box-shadow" href="{% if i > 1 %}{{ site.paginate_path | replace: ':num', i | relative_url }}{% else %}{{ '/' | relative_url }}{% endif %}">{{ i }}</a>
```
Comment (cotes2020) [inline _includes/post-paginator.html:50]: I don't think `/page{num}/` is an illegal path, please undo the changes here

File: _layouts/archives.html [hunk 3]
```diff
@@ -5,7 +5,8 @@
 
 <div id="archives" class="pl-xl-2">
 
-{% for post in site.posts %}
+{% assign posts = site.posts | where_exp: "item", "item.hidden != true" %}
```
Comment (cotes2020) [inline _layouts/archives.html:8]: `hidden: true` only hides the post on the homepage pagination, if we want to hide a certain post site-wide, we should use `published: false` instead. So please revert the changes here.

**Review by cotes2020 — CHANGES_REQUESTED** [review]

File: _includes/post-paginator.html (hunk 2 above)
Comment (ZakKemble) [reply _includes/post-paginator.html:50]: /page{num}/ does not work if "paginate_path" in _config.yml is changed to "/page/:num/".

Comment (cotes2020) [reply _includes/post-paginator.html:50]: Is there any particular reason to set it to `/page:num/` in _\_config.yml_? The default one `/page:num/` is good enough.

Comment (ZakKemble) [reply _includes/post-paginator.html:50]: I used to use Wordpress which had the pages as `/page/x/` so to keep links and things working I also want to use the same paths in Jekyll.

File: _layouts/archives.html (hunk 3 above)
Comment (ZakKemble) [reply _layouts/archives.html]: Ah ok

File: _includes/post-paginator.html (hunk 2 above)
Comment (ZakKemble) [reply _includes/post-paginator.html:50]: And of course if someone else wants to change the page paths for whatever reason then it will just work...

File: _layouts/archives.html (hunk 3 above)
Comment (ZakKemble) [reply _layouts/archives.html]: Actually I'm not sure about this. Using `published` will completly skip generating the page, but what if I want a post that is hidden from the public but can still be accessed if if the URL is known? For this I would use `hidden`, but it still shows up in archives, categories and tags pages which makes it a bit pointless.

File: _includes/post-paginator.html (hunk 2 above)
Comment (cotes2020) [reply _includes/post-paginator.html:50]: Alright, got it

File: _layouts/archives.html (hunk 3 above)
Comment (cotes2020) [reply _layouts/archives.html]: > what if I want a post that is hidden from the public but can still be accessed if the URL is known?

We could indeed do this by modifying the source code to hide all links to hidden posts on the site, but don't forget that search engines still crawl these hidden posts and display them on the internet. So, what's the point of doing this (`hidden: true`)?

I think the best way to hide a post is to not generate it or encrypt the post content, what do you think?

Comment (ZakKemble) [reply _layouts/archives.html]: If there are no links to the hidden post then search engines will not find it. It will also need to be hidden from feeds and sitemaps. I suppose it does seem like a pain to do properly since other feed and sitemap plugins don't really support hiding things.

**Comment by cotes2020** [comment]
This PR looks a bit chaotic, please make the following changes to the commits:

1. Drop f18d734d3deab29b2c40b49ba1a95dc43faa2ca7 and 07f3a9fa318d157dfc92d6c87de1eda8450382a3 
2. Merge 87743671ad731f7fff22ada67fccd90783c91c5d and 99a0de59ecccafcd61d57f25af4f6218397e68e2 into a single commit.

Each PR should only focus on one issue, too many commits mixed together can make issue tracking difficult, hope to understand. Anyway, thank you very much for your contribution to the project.

**Comment by cotes2020** [comment]
Sorry, I missed one thing: 3ecb2e915481f7d126419b7e016ce985a73e52b8 and 327f5ad26d6020fb377b4ab5a4424ddabb85d205 should also be combined into one commit
